utils/benchmarking/eventing/timeseriesdb/timeseriesdb_test.go

```diff
@@ -0,0 +1,137 @@
+// MIT License
+//
+// Copyright (c) 2021 Mert Bora Alper and EASE lab
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+package main
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	ctrdlog "github.com/containerd/containerd/log"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/ease-lab/vhive/utils/benchmarking/eventing/proto"
+)
+
+const SleepDuration = 200 * time.Millisecond
+
+func TestDefaultEndToEnd(t *testing.T) {
```

Comment:
this test function is too long. can you create few helper functions? also, consider splintering the test into several by a test objective?

Reply:
It is an integration test that simulates `chained-function-eventing`, and thus cannot be split into smaller tests as the order of operations are important. However, I tidied it up so hopefully looks much better now.